Gems/ROS2/Code/Source/SimulationUtils/FollowingCameraComponent.cpp

```diff
@@ -170,93 +194,323 @@ namespace ROS2
+    {
+        AZ::Vector2 mousePosition(0.0f, 0.0f);
+
+        // Query the input system for the current mouse position
+        AzFramework::InputSystemCursorRequestBus::EventResult(
+            mousePosition, AzFramework::InputDeviceMouse::Id, &AzFramework::InputSystemCursorRequests::GetSystemCursorPositionNormalized);
+
+        return mousePosition;
+    }
+
+    void FollowingCameraComponent::RotateCameraOnMouse(const AZ::Vector2& mouseDelta)
+    {
+        const float rotationSensitivity = m_configuration.m_rotationSensitivity;
+
+        // Convert mouse delta to rotation angles
+        float yawRotation = -mouseDelta.GetX() * rotationSensitivity; // Inverted X for reversed yaw rotation
+        float pitchRotation = -mouseDelta.GetY() * rotationSensitivity; // Inverted Y for pitch, adjust as needed
+
+        // Fetch the current camera orientation from m_cameraOffset
+        AZ::Quaternion currentRotation = AZ::Quaternion::CreateFromMatrix4x4(m_cameraOffset);
+
+        // Global Z-axis for yaw rotation
+        AZ::Vector3 globalZAxis = AZ::Vector3(0, 0, 1);
+        AZ::Quaternion yawQuat = AZ::Quaternion::CreateFromAxisAngle(globalZAxis, yawRotation);
+
+        // Apply yaw rotation to current camera orientation
+        AZ::Quaternion tempRotation = yawQuat * currentRotation;
+        tempRotation.Normalize();
+
+        // Calculate pitch rotation around the camera's local X-axis after applying yaw
+        // This ensures that pitch adjustments are made relative to the camera's adjusted orientation
+        AZ::Vector3 localXAxis = tempRotation.TransformVector(AZ::Vector3(1, 0, 0));
+        AZ::Quaternion pitchQuat = AZ::Quaternion::CreateFromAxisAngle(localXAxis, pitchRotation);
+
+        // Apply pitch rotation
+        AZ::Quaternion newRotation = pitchQuat * tempRotation;
+        newRotation.Normalize();
+
+        // Update the camera's offset matrix with the new orientation, keeping the position unchanged
+        AZ::Vector3 currentPosition = m_cameraOffset.GetTranslation();
+        m_cameraOffset = AZ::Matrix4x4::CreateFromQuaternionAndTranslation(newRotation, currentPosition);
+    }
+
+    void FollowingCameraComponent::MoveCameraOnKeys()
+    {
+        AZ::Matrix4x4 localCamPoseChange = AZ::Matrix4x4::CreateIdentity();
+        float translationSpeed = m_configuration.m_translationSpeed;
+
+        // Check each key state and adjust the camera pose change accordingly
+        if (m_keyStates[AzFramework::InputDeviceKeyboard::Key::AlphanumericW])
         {
-            m_rotationOffset -= m_configuration.m_rotationSpeed;
-            return;
+            localCamPoseChange = localCamPoseChange * AZ::Matrix4x4::CreateTranslation(AZ::Vector3::CreateAxisY() * translationSpeed);
         }
-        if (channelId == AzFramework::InputDeviceKeyboard::Key::AlphanumericD)
+        if (m_keyStates[AzFramework::InputDeviceKeyboard::Key::AlphanumericS])
```

Comment:
consider asserting on m_keyStates containing AzFramework::InputDeviceKeyboard::Key::AlphanumericS.
Please consider using/aliasing those long namespacing.

Reply:
Added alias. What would be the benefit of using an assert here?